src/sanity/sanity-checks.ts

```diff
@@ -0,0 +1,271 @@
+    }
+    if (response.status >= 300) {
+      throw Error(`Unable to get ${items} - status ${response.status} (${response.text})`)
+    }
+  }
+  const checks = [
+    check('Get calendar list', async () => {
+      await callGoogleApi(
+        `/www.googleapis.com/calendar/v3/users/${testUser}/calendarList`,
+        'https://www.googleapis.com/auth/calendar.readonly',
+        'calendars'
+      )
+    }),
+    check('Get messages list', async () => {
+      await callGoogleApi(
+        `/www.googleapis.com/gmail/v1/users/${testUser}/messages`,
+        'https://www.googleapis.com/auth/gmail.readonly',
+        'messages'
+      )
+    }),
+  ]
+  return checkSuite('GSUITE', checks)
+}
+
+const o365Suite = (app, module, testUser) => {
+  const skipIfCannotCheck = () => {
+    if (!module.enabled) {
+      throw new Skipped('Microsoft Graph module not enabled.')
+    }
+    if (!testUser) {
+      throw new Skipped('O365 test user not set (O365_TEST_USER).')
+    }
+  }
+  const callGraphApi = async (endpoint, items) => {
+    skipIfCannotCheck()
+    const apiToken = await appConfig.apiToken
+    const client = request(app)
+    let response
+    try {
+      response = await client.get(endpoint).set('Authorization', `Bearer ${apiToken}`)
+    } catch (err) {
+      throw Error(`Unable to get ${items}. See below for details:\n${err.message}`)
+    }
+    if (response.status >= 300) {
+      throw Error(`Unable to get ${items} - status ${response.status} (${response.text})`)
+    }
+  }
+  const checks = [
+    check('Get calendar list', async () => {
+      await callGraphApi(`/graph.microsoft.com/v1.0/users/${testUser}/calendars`, 'calendars')
+    }),
+    check('Get messages list', async () => {
+      await callGraphApi(`/graph.microsoft.com/v1.0/users/${testUser}/messages`, 'messages')
+    }),
+    check('Get events list', async () => {
+      await callGraphApi(`/graph.microsoft.com/v1.0/users/${testUser}/events`, 'events')
+    }),
+    check('Get mail folders', async () => {
+      await callGraphApi(`/graph.microsoft.com/v1.0/users/${testUser}/mailFolders`, 'mail folders')
+    }),
+  ]
+  return checkSuite('O365', checks)
+}
+
+export const runInitChecks = async (app) => {
+  const googleApiModule = await googleapis()
+  const gsuiteTestUser = process.env.GSUITE_TEST_USER
+  const msGraphModule = await microsoftgraph()
+  const o365TestUser = process.env.O365_TEST_USER
+
+  const suites = [
+    configSuite(),
+    appSuite(app),
+    gsuiteSuite(app, googleApiModule, gsuiteTestUser),
+    o365Suite(app, msGraphModule, o365TestUser)
+  ]
+
+  printHeader('RUNNING CHECKS', '=')
+  let abort = false
+  let total = 0
+  let skipped = 0
+  let failed = 0
+  for (let suite of suites) {
+    printHeader(`${suite.name}`, '-')
+    for (let check of suite.checks) {
+      await check.fn().then(() => {
+        console.log(`• ${check.name} - \x1b[32mOK\x1b[0m`)
```

Comment:
Color output should be disabled by default, it's usually useless in the cloud enviroment.

Reply:
removed